account_move_invoice_partner/models/account_move.py

```diff
@@ -0,0 +1,27 @@
+import logging
+
+from odoo import api, fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    partner_invoice_id = fields.Many2one(
+        "res.partner",
+        string="Invoice Address",
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]",
+    )
+
+    @api.onchange("partner_id")
```

Comment:
The onchange pattern is gone for Odoo 16. This needs refactoring to use the compute function.

Reply:
@janikvonrotz can you review this PR again please? I am not able to re-request